tests/tests_core/test_outliers.py

```diff
@@ -1,14 +1,75 @@
 from __future__ import division
-from expan.core.experiment import Experiment
+from expan.core.experiment import Experiment, _choose_threshold_type
 import pandas as pd
 import numpy as np
 
-def test_quantile_filtering():
+
+def test_choose_threshold_type_upper():
+    data = np.array([1, 2, 3])
+    assert _choose_threshold_type(data) == 'upper'
+
+
+def test_choose_threshold_type_upper_zero():
+    data = np.array([0, 1, 2, 3])
+    assert _choose_threshold_type(data) == 'upper'
+
+
+def test_choose_threshold_type_lower():
+    data = np.array([-3, -2, -1])
+    assert _choose_threshold_type(data) == 'lower'
+
+
+def test_choose_threshold_type_lower_zero():
+    data = np.array([-3, -2, -1, 0])
+    assert _choose_threshold_type(data) == 'lower'
+
+
+def test_choose_threshold_type_two_sided():
+    data = np.array([-3, -2, -1, 0, 1, 2, 3])
+    assert _choose_threshold_type(data) == 'two-sided'
+
+
+def test_quantile_filtering_upper_old():
     exp = Experiment({})
-    df = pd.DataFrame.from_dict({   'earnings' : np.array([0,0,1,2]) / np.array([0,0,1,1]) })
+    data = np.array([0,0,1,2]) / np.array([0,0,1,1])
+    df = pd.DataFrame.from_dict({'earnings' : data})
 
-    flags = exp._quantile_filtering(df, ['earnings'], 90, 'upper')
+    flags = exp._quantile_filtering(df, ['earnings'], {'earnings': ('upper', 90.0)})
     assert flags.tolist() == [False, False, False, True]
 
-    flags = exp._quantile_filtering(df, ['earnings'], 10, 'lower')
+
+def test_quantile_filtering_lower_old():
+    exp = Experiment({})
+    data = np.array([0,0,1,2]) / np.array([0,0,1,1])
+    df = pd.DataFrame.from_dict({'earnings' : data})
+
+    flags = exp._quantile_filtering(df, ['earnings'], {'earnings': ('lower', 10.)})
     assert flags.tolist() == [False, False, True, False]
+
+
+def test_quantile_filtering_upper():
+    exp = Experiment({})
+    data = np.array([0.0]*2 + list(range(10))) / np.array([0.0]*2 + [1.0]*10)
+    df = pd.DataFrame.from_dict({'earnings' : data})
+
+    flags = exp._quantile_filtering(df, ['earnings'], {'earnings': ('upper', 90.0)})
+    assert flags.tolist() == [False]*11 + [True]
+
+
+def test_quantile_filtering_lower():
+    exp = Experiment({})
+    data = np.array([0.0]*2 + list(range(10))) / np.array([0.0]*2 + [1.0]*10)
+    df = pd.DataFrame.from_dict({'earnings' : data})
+
+    flags = exp._quantile_filtering(df, ['earnings'], {'earnings': ('lower', 50.0)})
+    print(flags.tolist())
+    assert flags.tolist() == [False]*2 + [True]*5 + [False]*5
+
+
+def test_quantile_filtering_two_sided():
+    exp = Experiment({})
+    df = pd.DataFrame.from_dict({'earnings' : list(range(10))})
+
+    flags = exp._quantile_filtering(df, ['earnings'], {'earnings': ('two-sided', 80.0)})
+    results = flags.tolist()
+    assert results == [True] + [False]*8 + [True]
```

Comment:
In Google Chat, you said _"the asymmetric case is not yet implemented"_

This test tells me otherwise

Reply:
I forgot to add a smiley there !!

Have I misunderstood something?

Reply:
That's the symmetric case.

By saying symmetric, I mean that 0.5% of the whole distribution are cropped off on the right and the left side.

Asymmetric means that 0.5% of non-positive values are cropped off on the left and 0.5% of non-negative are cropped off on the right.

Reply:
Resolving. I see now that I had mixed up symmetry and assymetry